Dockerfile

```diff
@@ -1,14 +1,12 @@
-# latest kubectl in the SDK
-# https://cloud.google.com/sdk/docs/release-notes
-FROM google/cloud-sdk:alpine
-
-# Install kubectl
-RUN gcloud components install kubectl && \
-    rm -rf ./google-cloud-sdk/.install
+# see https://hub.docker.com/r/google/cloud-sdk/tags for available tags / versions
+ARG GCLOUD_SDK_VERSION
 
+FROM google/cloud-sdk:${GCLOUD_SDK_VERSION} AS cloud-sdk
```

Comment:
@tonglil 

yeah so this is something i spun my wheels on for a bit.. originally, i began down a different path based on the following from their [`README`](https://github.com/GoogleCloudPlatform/cloud-sdk-docker/blob/master/README.md#installing-different-version-of-gcloud-sdk):

> ### Installing different version of gcloud sdk:
> ```
> docker build -t my-cloud-sdk-docker:alpine --build-arg CLOUD_SDK_VERSION=<release_number> .
> ```

⚠️ note the difference between their name (`CLOUD_SDK_VERSION`) and the one used here (`GCLOUD_SDK_VERSION`)

turns out, that doesn't actually work 😞  (have a note to self to create an issue on their repo):

```txt
[Warning] One or more build-args [CLOUD_SDK_VERSION] were not consumed
```

then i realized we could probably get most of the way there using our own build `ARG` .. ok, simple enough, right?

... currently, our `:latest` / develop builds use the upstream `:alpine` tag:

https://github.com/nytimes/drone-gke/blob/b55353f12763e61f2948172e6416dd911e21a796/Dockerfile#L3

and so i thought of using _that_ as the default:

```dockerfile
ARG GCLOUD_SDK_VERSION alpine
FROM google/cloud-sdk:${GCLOUD_SDK_VERSION}
# ...
```

but then to support custom versions, this becomes a bit less straight forward:

```sh
docker build --build-arg GCLOUD_SDK_VERSION=237.0.0 .
```

when replaced in the Dockerfile, we'd end up building from their debian based default image instead of the alpine based image:

```dockerfile
# after replacement from `docker build --build-arg GCLOUD_SDK_VERSION=237.0.0`
FROM google/cloud-sdk:237.0.0
# ...
```

to get the desired outcome, we'd have to run something like:

```sh
docker build --build-arg GCLOUD_SDK_VERSION=237.0.0-alpine .
```

at which point we're specifying the entire tag, not just the version :thinking:

... so maybe we address that using some simple bash modifiers?

```dockerfile
# the gcloud SDK release version (e.g., 254.0.0)
ARG GCLOUD_SDK_VERSION

# the google/cloud-sdk image variant (e.g., alpine)
ARG GCLOUD_SDK_VARIANT alpine

# if GCLOUD_SDK_VERSION is set
# add a '-' between it and GCLOUD_SDK_VARIANT
ARG GCLOUD_SDK_TAG_SEPARATOR ${GCLOUD_SDK_VERSION:+'-'}

# if GCLOUD_SDK_VERSION is _not_ set
# include only the GCLOUD_SDK_VARIANT in the tag (e.g. ':slim')
# otherwise, include GCLOUD_SDK_VERSION and GCLOUD_SDK_VARIANT
# separated by GCLOUD_SDK_TAG_SEPARATOR
ARG GCLOUD_SDK_TAG ${GCLOUD_SDK_VERSION:-${GCLOUD_SDK_VARIANT}}${GCLOUD_SDK_TAG_SEPARATOR}${GCLOUD_SDK_VERSION:+${GCLOUD_SDK_VARIANT}}

FROM google/cloud-sdk:${GCLOUD_SDK_VERSION}
# ...
```

... then i thought "while i'm at it with all these build args, why not support the non-alpine variants? and what about `:latest` ?!" :man_facepalming: :

```dockerfile
# the gcloud SDK release version (e.g., 254.0.0)
ARG GCLOUD_SDK_VERSION

# the google/cloud-sdk image variant (e.g., alpine)
ARG GCLOUD_SDK_VARIANT

# if GCLOUD_SDK_VERSION and GCLOUD_SDK_VARIANT are both set
# add a '-' between GCLOUD_SDK_VERSION and GCLOUD_SDK_VARIANT
ARG GCLOUD_SDK_TAG_SEPARATOR ${GCLOUD_SDK_VERSION:+${GCLOUD_SDK_VARIANT:+'-'}}

##
# the moment you realize the Dockerfile syntax was
# not designed for use cases like this ...
#
#   if both GCLOUD_SDK_VERSION and GCLOUD_SDK_VARIANT are set
#   include each separated by a dash:
#
#   ```sh
#   docker build \
#     --build-arg 'GCLOUD_SDK_VERSION=254.0.0' \
#     --build-arg 'GCLOUD_SDK_VARIANT=alpine' .
#
#   # => FROM google/cloud-sdk:254.0.0-alpine
#   ```
#
#   if neither GCLOUD_SDK_VERSION nor GCLOUD_SDK_VARIANT are set
#   default to 'latest':
#
#   ```sh
#   docker build .
#
#   # => FROM google/cloud-sdk:latest
#   ```
#
#   if GCLOUD_SDK_VERSION is _not_ set but GCLOUD_SDK_VARIANT _is_
#   use only GCLOUD_SDK_VARIANT:
#
#   ```sh
#   docker build \
#     --build-arg 'GCLOUD_SDK_VARIANT=slim' .
#
#   # => FROM google/cloud-sdk:slim
#   ```
##
ARG GCLOUD_SDK_TAG ${GCLOUD_SDK_VERSION:-${GCLOUD_SDK_VARIANT:-'latest'}}${GCLOUD_SDK_TAG_SEPARATOR}${GCLOUD_SDK_VERSION:+${GCLOUD_SDK_VARIANT}}

FROM google/cloud-sdk:${GCLOUD_SDK_TAG} AS cloud-sdk
# ...
```

somewhere along the way, i reverted back to working off the original (single `GCLOUD_SDK_VERSION` build arg) and ultimately decided it would probably be less confusing to always explicitly specify the "version", but i never got around to renaming this build arg to `GCLOUD_SDK_TAG` :man_shrugging:

_so_ i can either rename that `GCLOUD_SDK_VERSION` to `GCLOUD_SDK_TAG` (or any other name you'd prefer), _or_ i can revisit those more complex options .. wdyt? (cc @fsouza )

Reply:
i just renamed this to `GCLOUD_SDK_TAG`

Reply:
It's a hole that turtles down :P 

I see no reason to build this image with anything other than the alpine version since no other tools in the sdk are used, and alpine is slimmer than slim.

The way you have it now LGTM!